### PR TITLE
Switch to Individual Prometheus / Grafana Charts

### LINKED ISSUE #166: Update prometheus-operator deployments
#155 and #158 set up the `prometheus-operator` deployment. 

However, `prometheus-operator` is now [deprecated](https://github.com/helm/charts/tree/master/stable/prometheus-operator#%EF%B8%8F--deprecated). The proper Helm chart is now [`kube-prometheus-stack`](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack). This should be changed. Hopefully it's as simple as changing the repo / chart names. There is also a new `CRD`: `probes.monitoring.coreos.com`.

Additionally, there will have to be a little work to enable HTTPS for the Grafana endpoint, see discussion here: https://github.com/pangeo-data/pangeo-binder/pull/155#issuecomment-670199494. 

If we want to make setup easier, we should configure Grafana to use its own LoadBalancer. This will cost money. If we want to avoid that, we will need to figure out how to set up another `Issuer` that uses the [DNS01 Solver](https://cert-manager.io/docs/configuration/acme/dns01/) (as opposed to the [HTTP01 Ingress Provider](https://cert-manager.io/docs/configuration/acme/http01/), which we use in binderhubs, ex. [AWS staging issuer](https://github.com/pangeo-data/pangeo-binder/blob/staging/k8s-aws/binderhub-issuer-staging.yaml#L17) and [GCP staging issuer](https://github.com/pangeo-data/pangeo-binder/blob/staging/deploy/binderhub-issuer-staging.yaml#L17)).

### PR DESCRIPTION
**Pull request by salvis2**
Closes #166 . See that issue for background on why the other helm charts are no good.

The old `stable` helm charts for Prometheus and Grafana have been deprecated, the new ones are here:
- [Grafana](https://github.com/grafana/helm-charts/tree/main/charts/grafana)
- [Prometheus](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus)

`pangeo-binder/requirements.yaml` and `k8s-aws/readme.md` have been updated with the new helm repos. Both charts are pinned to their most recent versions. 

Configuration changes were mostly untabbing, since there is no longer an umbrella chart for these two. However, I changed Grafana's ingress configuration from the example in the [Grafana readme](https://github.com/grafana/helm-charts/tree/main/charts/grafana#how-to-serve-grafana-with-a-path-prefix-grafana), and suddenly it works with HTTPS. I also had to manually fill in the location of the Prometheus data source. The only way I could get that connected was by using the `ClusterIP` I find with `kubectl get svc -n staging staging-prometheus-server`. I moved the datasource configuration to the secrets file, since now it will depend on the cluster and should not be a publicly-known address.

For now, since we only need the one deployment, I will leave it on `staging` and remove the AWS `prod` config for monitoring. @TomAugspurger let me know if you'd like me to set up the equivalent config for GCP. I should be able to log in and test manual deployment on staging / get the `ClusterIP` myself.

### CODE REVIEW

**Comment by salvis2** [comment]
I will also update the CI and fix that merge conflict.

File: deploy/prod.yaml [hunk 1]
```diff
@@ -86,7 +86,7 @@ nginx-ingress:
     service:
       loadBalancerIP: 35.226.186.40
 
-prometheus-operator:
+kube-prometheus-stack:
```
Comment (TomAugspurger) [inline deploy/prod.yaml]: So, do these config values have any effect now? I'd think they'd only have an effect if we're using the `kube-prometheus-stack` helm chart. But IIUC we aren't using that chart? We're using `prometheus` and `grafana` separately?

Comment (salvis2) [reply deploy/prod.yaml]: They wouldn't. I hadn't changed GCP values aside from that first commit that changed the umbrella chart, but can do that next. Just wanted the go-ahead from you first.

Comment (TomAugspurger) [reply deploy/prod.yaml]: Thanks, I'm happy to do it all in one go here.

Comment (TomAugspurger) [reply deploy/prod.yaml]: Let me know if there's anything I need to do on the GCP side (IP addresses or anything like that).

Comment (salvis2) [reply deploy/prod.yaml]: I should be able to authenticate into the GCP binder. I'll manually deploy staging and get the IP to make sure it all works. Will need an extra PR to plug in the prometheus IP address for prod if we don't want to deploy prod manually (which is probably the case).

**Comment by salvis2** [comment]
Monitoring deployments are up!
- [AWS staging Grafana site](https://staging.aws-uswest2-binder.pangeo.io/grafana/?orgId=1)
- [GCP staging Grafana site](https://staging.binder.pangeo.io/grafana/?orgId=1)

Both have HTTPS, which is cool. However, the GCP site seems to have almost no data. If I launch the dashboard "Cluster Monitoring for Kubernetes" (you should see it if you go to Dashboards > Manage), I find many pods with the AWS graphs, but only one on GCP. That single "pod" is named "Value" but is using quite a bit of memory, so I'm suspecting this is just the total.

A thought that I had when looking at the GCP binder cluster: I think there's a config error with some ingress bit on staging. I see `kubectl get svc -n staging binderhub-proxy-nginx-ingress-controller` has an `EXTERNAL-IP` of `<pending>`. There is also a Service `binderhub-proxy-nginx-ingress-default-backend`. There are similar Services but with names that start with `staging` or `prod` instead of `binderhub-proxy` on their respective namespaces. The proper services are also on AWS.

On both staging deployments, I ran `kubectl get deployments -n staging`. The GCP one appears to have these extra `nginx-ingress` bits that are 41 days old. Maybe they were manually deployed by accident? Do you think I can just delete them @TomAugspurger ?

**Comment by TomAugspurger** [comment]
It's definitely possible that I messed up the ingress stuff.  I'm fine with you deleting them, and if something breaks then I can take a look.

**Comment by salvis2** [comment]
Deleted those Services / Deployments and nothing broke. The dashboards still have wildly different levels of details though.

**Comment by TomAugspurger** [comment]
Thanks for working on this. Feel free to merge when you're ready.

**Pull request merged**